docs/content/sips/008-logging.md

```diff
@@ -0,0 +1,104 @@
+title = "SIP 008 - Runtime Logging"
+template = "main"
+date = "2022-12-09T11:36:36Z"
+---
+
+Summary: Unified runtime logging
+
+Owner: mail@etehtsea.me
+
+Created: December 09, 2022
+
+## Background
+
+## Current experience
+
+Currently, by default spin prints into `stdout` only the basic info:
+
+```
+$ spin up
+
+Serving http://127.0.0.1:3000
+Available Routes:
+  hello: http://127.0.0.1:3000/hello
+    A simple component that returns hello.
+```
+
+Application log is implicitly written into `~/.spin/<spin_app>/logs/<component_id>_std{err|out}.log`.
+The log for the spin itself has log level `ERROR` and outputs into `stderr`.
+
+To enable a full logging experience, a developer needs to provide `--follow-all` to duplicate app log into `stderr` and also `RUST_LOG=spin=<log_level>` to see the spin log in the `stderr`.
+
+
+## Proposal
+
+- Set default log level to `INFO`.
+- Set default log destination to `stdout/stderr`.
```

Comment:
> This is particularly the case for commands like spin deploy or spin templates list where there is salient output that we are trying to show to the user (the deployment URL, the list) and we don't want that to be just another logging message amongst dozens.

That's right, these command shouldn't be affected. The only case for commands like these is to additionally provide something like `--debug` to enable debug/trace logging.

> The combination of these two feels like it could get extremely noisy. For myself, I'd prefer the default experience to be fairly minimal, and to opt into a more verbose flow as and when I want it. I may be an outlier here though!

If `INFO` logging gets too noisy then it might make sense to review what is actually logged with this log level and change some of the output to `debug`/`trace`.

> That may mean we need different default behaviour for the trigger commands (spin up) vs the non-trigger commands. Which is fine, of course, just needs to be in our minds.

Yep, that's why I narrowed it down to "runtime" logging but didn't provide the description of what I meant by it. Will adjust the doc.